### DISCUSSION TITLE
Segment marked as "bad"

### DISCUSSION THREAD

**Discussion by kimelit** [raphaelvallat/yasa#133] [Q&A] [answered]
Hello :)
The _yasa.bandpower_ function ignores the "BAD"/"bad" annotations, right?
For example, if I have a segment annotated as N2, and N2 is in the _include_ list, then the entire segment will be included in the band-power calculation, although part of it is marked a "bad", right?
Thanks!

**Comment by raphaelvallat** [comment, marked as answer]
Hi —That's correct. Where is the "Bad" annotation coming from? Manual or automatic input?

**Reply by kimelit** [reply]
Thank you for the reply.

It's from a manual marking of noise. Does the origin matter?
I've started to write a script that will eliminate all overlaps. It is a bit of a headache because of all the possible cases, but I guess that I don't have a choice, right?
For spindle detection (I'm using yasa for this as well, thank you!) I think it's not important, but for power analysis it might be problematic to use noise as part of the data. Right?

Thanks!

**Reply by raphaelvallat** [reply]
The origin does not matter, I was just curious. The bandpower estimation in YASA should be pretty robust to "noise bursts" (e.g. artifacts, motion) in the data. What's more problematic however is when you have a channel that's consistently bad/noisy for most of the night, in which case I would recommend excluding this particular channel from the analyses.

**Comment by kimelit** [comment]
Hi,
Thank you :)

The noisy channel will only have an effect on itself, right? And not on other channels?

Could you perhaps briefly mention why local strong noise will not hamper the power estimation?

Also, thinking again - can such noise cause the algorithm to "miss" spindles? Perhaps due to erroneous baseline-voltage estimation?

Thank you!

**Reply by raphaelvallat** [reply]
Hi,

> The noisy channel will only have an effect on itself, right? And not on other channels?

That's correct.

> Could you perhaps briefly mention why local strong noise will not hamper the power estimation?

YASA uses a **_median_**-averaging of the sliding periodograms in scipy.stats.welch instead of the default mean-averaging. The former is more robust to noise and artefacts.

> Also, thinking again - can such noise cause the algorithm to "miss" spindles?

It really depends on the nature of the noise, and the frequencies that it affects. But yeah it could. You will get better results if you apply the spindles detection on artefact-free NREM sleep.

Thanks

**Comment by kimelit** [comment]
Thank you :)
How do users create an artefact-free data without this double-marking that I described? I've used the most standard mne procedures...

**Comment by kimelit** [comment]
Jus to let you know that I've written a script that removes the overlaps :) So if that's of interest, would be happy to share it of course.

**Reply by raphaelvallat** [reply]
Please feel free to share here!
